Sources/AsyncHTTPClient/HTTPHandler.swift

```diff
@@ -277,25 +209,28 @@ extension HTTPClient {
 
         /// Resolved port.
         public var port: Int {
-            switch self.connectionTarget {
+            switch self.deconstructedURL.connectionTarget {
             case .ipAddress(_, let address): return address.port!
             case .domain(_, let port): return port
-            case .unixSocket: return Request.useTLS(self.scheme) ? 443 : 80
+            case .unixSocket: return self.deconstructedURL.scheme.defaultPort
             }
         }
 
+        /// Whether request will be executed using secure socket.
+        public var useTLS: Bool { self.deconstructedURL.scheme.useTLS }
```

Comment:
I think `uses` is better.

Reply:
Also, why do we add this public API.

Reply:
It’s already public api and we can’t change the naming

Reply:
Oh I see. Makes sense!